pytorch_lightning/utilities/auto_restart.py

```diff
@@ -564,35 +563,71 @@ def _add_capture_metadata_collate(dataloader: DataLoader) -> None:
     )
 
 
-def reload_dataloader_state_dict(dataloader: DataLoader, state_dict: Dict[str, Any]) -> None:
+def _reload_dataloader_state_dict(dataloader: DataLoader, state_dict: Dict[str, Any]) -> None:
```

Comment:
For the sake of good code practices, this should be

```python
# loop.py
ft_mode = self.trainer.state._fault_tolerant_mode
if ft_mode.is_automatic:
    _reload_automatic_dataloader_state()
elif ft_mode.is_manual:
    _reload_manual_dataloader_state()
```

Instead of checking the mode inside and have two completely different code paths for each mode. This aids in testing and separates responsabilities.

Reply:
Done !